### ISSUE TITLE
Complains we are writing to two different databases in the same transaction

### ISSUE THREAD

**Issue by BugGambit** [neo4j/neo4j#12705]
## Environment info
- Neo4j version: 4.2.1-enterprise (Docker)
- Operating system: Container-Optimized OS with Docker (cos)
- API/Driver:  neo4j-driver (js) 4.1.2
- **Steps to reproduce**: We are not able to reproduce the error consistently.

## Logs

```
Writing to more than one database per transaction is not allowed.
Attempted write to Local{graphId=180, uuid=156c3b1e-081b-4479-b4f1-46d45e377174, databaseName='admin'},
currently writing to Local{graphId=179, uuid=156c3b1e-081b-4479-b4f1-46d45e377174, databaseName='admin'}
```

It's hard to make sense of the error message as the `databaseName` and `uuid` equals in those two statements, but the `graphId` does not.

We are utilizing multi-database support in Neo4J and the `admin` database is one of the databases we have created.

**Comment by mnd999** [comment]
Thanks for the report, I think I've been able to reproduce the issue by creating a new database concurrently with a transaction performing multiple writes. Is it possible this is the issue you have observed?

**Comment by BugGambit** [comment]
@mnd999 Can you describe that further?
Is this correct: a transaction `A` creates a new database `X` while transaction `B` is performing multiple writes to another database `Y`?

**Comment by mnd999** [comment]
Yes, that's correct. The create database has to occur between two writes in the other transaction as far as I can tell. This is definitely a bug and I will merge a fix for it which will go into the next 4.2.x release. 

I'm hoping this is the same issue you are observing so I want to check whether you have operations affecting the database catalog going on when you saw the errors. If you did, it's likely to be the same thing.

**Comment by mnd999** [comment]
Should be resolved by bb5542e2e4ea3d3ea0df693bfc9182591805de7d / Neo4j 4.2.6

**Comment by BugGambit** [comment]
Thanks
